iOS - build fails if using "WKWebViewOnly" preference

# Bug Report

## Problem

### What is expected to happen?
`cordova build ios` Build success


### What does actually happen?
```
$ cordova build ios
...
** BUILD FAILED **


The following build commands failed:
        CompileC /Users/foo/Library/Developer/Xcode/DerivedData/HelloWorld-euwessldwjxdluatamnrumldwksu/Build/Intermediates.noindex/HelloWorld.build/Debug-iphonesimulator/HelloWorld.build/Objects-normal/x86_64/CDVUIInAppBrowser.o /Users/foo/dev/hello/platforms/ios/HelloWorld/Plugins/cordova-plugin-inappbrowser/CDVUIInAppBrowser.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
xcodebuild: Command failed with exit code 65
```


## Information
<!-- Include all relevant information that might help understand and reproduce the problem -->
Since [cordova-ios@5.1.0](https://cordova.apache.org/announcements/2019/11/25/cordova-ios-release-5.1.0.html) and following https://github.com/apache/cordova-ios/pull/715
If you add "WKWebViewOnly" preference, then `cordova build` fails on using cordova-plugin-inappbrowser


(ping https://github.com/apache/cordova-plugin-inappbrowser/pull/535 & https://github.com/apache/cordova-plugin-inappbrowser/issues/532)


### Command or Code
<!-- What command or code is needed to reproduce the problem? -->
```
cordova create hello com.example.hello HelloWorld
cd hello
cordova platform add ios
cordova plugin add cordova-plugin-inappbrowser
<<add "WKWebViewOnly" preference into config.xml>>
cordova build ios
```


### Environment, Platform, Device
<!-- In what environment, on what platform or on which device are you experiencing the issue? -->



### Version information
<!-- 
What are relevant versions you are using?
For example:
Cordova: Cordova CLI, Cordova Platforms, Cordova Plugins 
Other Frameworks: Ionic Framework and CLI version
Operating System, Android Studio, Xcode etc.
-->
Cordova CLI *9.0.0*
Cordova Platform *iOS* (5.1.0)
Cordova Plugin *cordova-plugin-inappbrowser* (3.1.0)


## Checklist
<!-- Please check the boxes by putting an x in the [ ] like so: [x] -->

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

True, inappbrowser still uses UIWebview in some parts of the code

`CompileC /Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/IntermediateBuildFilesPath/Ondrowa.build/Release-iphoneos/Ondrowa.build/Objects-normal/armv7/CDVInAppBrowser.o Ondrowa/Plugins/cordova-plugin-inappbrowser/CDVInAppBrowser.m normal armv7 objective-c com.apple.compilers.llvm.clang.1_0.compiler
    cd /Users/vagrant/git/platforms/ios
    export LANG=en_US.US-ASCII
    export PATH="/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode-beta.app/Contents/Developer/usr/bin:/Users/vagrant/.bitrise/tools:/Users/vagrant/.jenv/shims:/Users/vagrant/.rbenv/shims:/usr/local/bin:/usr/local/sbin:~/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/vagrant/.jenv/shims:/Users/vagrant/.rbenv/shims:/usr/local/sbin:/Users/vagrant/bin:/usr/local/opt/go/libexec/bin:/Users/vagrant/go/bin:/Users/vagrant/bitrise/tools/cmd-bridge/bin/osx:/opt/android-ndk:/usr/local/share/android-sdk/tools:/usr/local/share/android-sdk/tools/bin:/usr/local/share/android-sdk/platform-tools:/usr/local/flutter/bin:/Users/vagrant/.jenv/bin:/usr/local/opt/go/libexec/bin:/Users/vagrant/go/bin:/Users/vagrant/bitrise/tools/cmd-bridge/bin/osx:/opt/android-ndk:/usr/local/share/android-sdk/tools:/usr/local/share/android-sdk/tools/bin:/usr/local/share/android-sdk/platform-tools:/usr/local/flutter/bin:/Users/vagrant/.jenv/bin"
    /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -target armv7-apple-ios10.0 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=c99 -fobjc-arc -fmodules -gmodules -fmodules-cache-path=/Users/vagrant/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/vagrant/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wno-trigraphs -fpascal-strings -Os -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Wunreachable-code -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wimplicit-retain-self -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -DWK_WEB_VIEW_ONLY=1 -DCOCOAPODS=1 -DWK_WEB_VIEW_ONLY=1 -DGPB_USE_PROTOBUF_FRAMEWORK_IMPORTS=1 -DWK_WEB_VIEW_ONLY=1 -DPB_FIELD_32BIT=1 -DPB_NO_PACKED_STRUCTS=1 -DPB_ENABLE_MALLOC=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.2.sdk -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -g -fvisibility=hidden -Wno-sign-conversion -Winfinite-recursion -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wno-semicolon-before-method-body -iquote /Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/IntermediateBuildFilesPath/Ondrowa.build/Release-iphoneos/Ondrowa.build/Ondrowa-generated-files.hmap -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/IntermediateBuildFilesPath/Ondrowa.build/Release-iphoneos/Ondrowa.build/Ondrowa-own-target-headers.hmap -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/IntermediateBuildFilesPath/Ondrowa.build/Release-iphoneos/Ondrowa.build/Ondrowa-all-non-framework-target-headers.hmap -ivfsoverlay /Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/IntermediateBuildFilesPath/Ondrowa.build/all-product-headers.yaml -iquote /Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/IntermediateBuildFilesPath/Ondrowa.build/Release-iphoneos/Ondrowa.build/Ondrowa-project-headers.hmap -I/Users/vagrant/git/platforms/ios/build/device/include -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/InstallationBuildProductsLocation/Applications/usr/local/lib/include -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/IntermediateBuildFilesPath/UninstalledProducts/include -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/include -I/Users/vagrant/git/platforms/ios/build/device -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/FirebaseABTesting/FirebaseABTesting.framework/Headers -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/FirebaseAuth/FirebaseAuth.framework/Headers -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/FirebaseCore/FirebaseCore.framework/Headers -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/FirebaseCoreDiagnostics/FirebaseCoreDiagnostics.framework/Headers -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/FirebaseInstanceID/FirebaseInstanceID.framework/Headers -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/FirebaseMessaging/FirebaseMessaging.framework/Headers -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/FirebaseRemoteConfig/FirebaseRemoteConfig.framework/Headers -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/GTMSessionFetcher/GTMSessionFetcher.framework/Headers -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/GoogleDataTransport/GoogleDataTransport.framework/Headers -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/GoogleDataTransportCCTSupport/GoogleDataTransportCCTSupport.framework/Headers -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/GoogleToolboxForMac/GoogleToolboxForMac.framework/Headers -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/GoogleUtilities/GoogleUtilities.framework/Headers -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/Protobuf/protobuf.framework/Headers -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/nanopb/nanopb.framework/Headers -I/Users/vagrant/git/platforms/ios/Pods/Headers/Public -I/Users/vagrant/git/platforms/ios/Pods/Headers/Public/Firebase -I/Users/vagrant/git/platforms/ios/Pods/Headers/Public/FirebaseAnalyticsInterop -I/Users/vagrant/git/platforms/ios/Pods/Headers/Public/FirebaseAuthInterop -I/Users/vagrant/git/platforms/ios/Pods/Headers/Public/FirebaseCoreDiagnosticsInterop -I/Users/vagrant/git/platforms/ios/Pods/Firebase/CoreOnly/Sources -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/IntermediateBuildFilesPath/Ondrowa.build/Release-iphoneos/Ondrowa.build/DerivedSources-normal/armv7 -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/IntermediateBuildFilesPath/Ondrowa.build/Release-iphoneos/Ondrowa.build/DerivedSources/armv7 -I/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/IntermediateBuildFilesPath/Ondrowa.build/Release-iphoneos/Ondrowa.build/DerivedSources -F/Users/vagrant/git/platforms/ios/build/device -F/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/FirebaseABTesting -F/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/FirebaseAuth -F/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/FirebaseCore -F/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/FirebaseCoreDiagnostics -F/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/FirebaseInstanceID -F/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/FirebaseMessaging -F/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/FirebaseRemoteConfig -F/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/GTMSessionFetcher -F/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/GoogleDataTransport -F/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/GoogleDataTransportCCTSupport -F/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/GoogleToolboxForMac -F/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/GoogleUtilities -F/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/Protobuf -F/Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/BuildProductsPath/Release-iphoneos/nanopb -F/Users/vagrant/git/platforms/ios/Pods/Crashlytics/iOS -F/Users/vagrant/git/platforms/ios/Pods/Fabric/iOS -F/Users/vagrant/git/platforms/ios/Pods/FirebaseAnalytics/Frameworks -F/Users/vagrant/git/platforms/ios/Pods/FirebasePerformance/Frameworks -F/Users/vagrant/git/platforms/ios/Pods/GoogleAppMeasurement/Frameworks -include /Users/vagrant/git/platforms/ios/build/sharedpch/Ondrowa-Prefix-awksqhvofjbsjxdhrkcbdagvaqxd/Ondrowa-Prefix.pch -MMD -MT dependencies -MF /Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/IntermediateBuildFilesPath/Ondrowa.build/Release-iphoneos/Ondrowa.build/Objects-normal/armv7/CDVInAppBrowser.d --serialize-diagnostics /Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/IntermediateBuildFilesPath/Ondrowa.build/Release-iphoneos/Ondrowa.build/Objects-normal/armv7/CDVInAppBrowser.dia -c /Users/vagrant/git/platforms/ios/Ondrowa/Plugins/cordova-plugin-inappbrowser/CDVInAppBrowser.m -o /Users/vagrant/Library/Developer/Xcode/DerivedData/Ondrowa-hakpcouewcxruugbaiotbdmxnayz/Build/Intermediates.noindex/ArchiveIntermediates/Ondrowa/IntermediateBuildFilesPath/Ondrowa.build/Release-iphoneos/Ondrowa.build/Objects-normal/armv7/CDVInAppBrowser.o
In file included from /Users/vagrant/git/platforms/ios/Ondrowa/Plugins/cordova-plugin-inappbrowser/CDVInAppBrowser.m:22:
/Users/vagrant/git/platforms/ios/Ondrowa/Plugins/cordova-plugin-inappbrowser/CDVUIInAppBrowser.h:64:5: error: unknown type name 'CDVUIWebViewDelegate'
    CDVUIWebViewDelegate* _webViewDelegate;
    ^
1 error generated.`


further reading:
https://ionicframework.com/blog/understanding-itms-90809-uiwebview-api-deprecation/
https://cordova.apache.org/announcements/2019/11/25/cordova-ios-release-5.1.0.html
https://github.com/apache/cordova-ios/pull/715

--- a/src/ios/CDVInAppBrowser.m
+++ b/src/ios/CDVInAppBrowser.m
@@ -19,7 +19,9 @@
 
 #import "CDVInAppBrowser.h"
 #import "CDVInAppBrowserOptions.h"
+#if !WK_WEB_VIEW_ONLY
 #import "CDVUIInAppBrowser.h"
+#endif
 #import "CDVWKInAppBrowser.h"
 #import <Cordova/CDVPluginResult.h>
 
@@ -49,85 +51,121 @@
         return;
     }
     self.usewkwebview = browserOptions.usewkwebview;
-    if(self.usewkwebview){
-        [[CDVWKInAppBrowser getInstance] open:command];
-    }else{
-        [[CDVUIInAppBrowser getInstance] open:command];
-    }
+    #if WK_WEB_VIEW_ONLY
+      [[CDVWKInAppBrowser getInstance] open:command];
+    #else
+      if(self.usewkwebview){
+          [[CDVWKInAppBrowser getInstance] open:command];
+      }else{
+          [[CDVUIInAppBrowser getInstance] open:command];
+      }
+    #endif
 }
 
 - (void)close:(CDVInvokedUrlCommand*)command
 {
-    if(self.usewkwebview){
-        [[CDVWKInAppBrowser getInstance] close:command];
-    }else{
-        [[CDVUIInAppBrowser getInstance] close:command];
-    }
+    #if WK_WEB_VIEW_ONLY
+      [[CDVWKInAppBrowser getInstance] close:command];
+    #else
+      if(self.usewkwebview){
+          [[CDVWKInAppBrowser getInstance] close:command];
+      }else{
+          [[CDVUIInAppBrowser getInstance] close:command];
+      }
+    #endif
 }
 
 
 - (void)show:(CDVInvokedUrlCommand*)command
 {
-    if(self.usewkwebview){
-        [[CDVWKInAppBrowser getInstance] show:command];
-    }else{
-        [[CDVUIInAppBrowser getInstance] show:command];
-    }
+    #if WK_WEB_VIEW_ONLY
+      [[CDVWKInAppBrowser getInstance] show:command];
+    #else
+      if(self.usewkwebview){
+          [[CDVWKInAppBrowser getInstance] show:command];
+      }else{
+          [[CDVUIInAppBrowser getInstance] show:command];
+      }
+    #endif
 }
 
 - (void)hide:(CDVInvokedUrlCommand*)command
 {
-    if(self.usewkwebview){
-        [[CDVWKInAppBrowser getInstance] hide:command];
-    }else{
-        [[CDVUIInAppBrowser getInstance] hide:command];
-    }
+    #if WK_WEB_VIEW_ONLY
+      [[CDVWKInAppBrowser getInstance] hide:command];
+    #else
+      if(self.usewkwebview){
+          [[CDVWKInAppBrowser getInstance] hide:command];
+      }else{
+          [[CDVUIInAppBrowser getInstance] hide:command];
+      }
+    #endif
 }
 
 
 - (void)injectScriptCode:(CDVInvokedUrlCommand*)command
 {
-    if(self.usewkwebview){
-        [[CDVWKInAppBrowser getInstance] injectScriptCode:command];
-    }else{
-        [[CDVUIInAppBrowser getInstance] injectScriptCode:command];
-    }
+    #if WK_WEB_VIEW_ONLY
+      [[CDVWKInAppBrowser getInstance] injectScriptCode:command];
+    #else
+      if(self.usewkwebview){
+          [[CDVWKInAppBrowser getInstance] injectScriptCode:command];
+      }else{
+          [[CDVUIInAppBrowser getInstance] injectScriptCode:command];
+      }
+    #endif
 }
 
 - (void)injectScriptFile:(CDVInvokedUrlCommand*)command
 {
-     if(self.usewkwebview){
-        [[CDVWKInAppBrowser getInstance] injectScriptFile:command];
-    }else{
-        [[CDVUIInAppBrowser getInstance] injectScriptFile:command];
-    }
+    #if WK_WEB_VIEW_ONLY
+      [[CDVWKInAppBrowser getInstance] injectScriptCode:command];
+    #else
+      if(self.usewkwebview){
+          [[CDVWKInAppBrowser getInstance] injectScriptCode:command];
+      }else{
+          [[CDVUIInAppBrowser getInstance] injectScriptCode:command];
+      }
+    #endif
 }
 
 - (void)injectStyleCode:(CDVInvokedUrlCommand*)command
 {
-    if(self.usewkwebview){
-        [[CDVWKInAppBrowser getInstance] injectStyleCode:command];
-    }else{
-        [[CDVUIInAppBrowser getInstance] injectStyleCode:command];
-    }
+    #if WK_WEB_VIEW_ONLY
+      [[CDVWKInAppBrowser getInstance] injectStyleCode:command];
+    #else
+      if(self.usewkwebview){
+          [[CDVWKInAppBrowser getInstance] injectStyleCode:command];
+      }else{
+          [[CDVUIInAppBrowser getInstance] injectStyleCode:command];
+      }
+    #endif
 }
 
 - (void)injectStyleFile:(CDVInvokedUrlCommand*)command
 {
-    if(self.usewkwebview){
-        [[CDVWKInAppBrowser getInstance] injectStyleFile:command];
-    }else{
-        [[CDVUIInAppBrowser getInstance] injectStyleFile:command];
-    }
+    #if WK_WEB_VIEW_ONLY
+      [[CDVWKInAppBrowser getInstance] injectStyleFile:command];
+    #else
+      if(self.usewkwebview){
+          [[CDVWKInAppBrowser getInstance] injectStyleFile:command];
+      }else{
+          [[CDVUIInAppBrowser getInstance] injectStyleFile:command];
+      }
+    #endif
 }
 
 - (void)loadAfterBeforeload:(CDVInvokedUrlCommand*)command
 {
-    if(self.usewkwebview){
-        [[CDVWKInAppBrowser getInstance] loadAfterBeforeload:command];
-    }else{
-        [[CDVUIInAppBrowser getInstance] loadAfterBeforeload:command];
-    }
+    #if WK_WEB_VIEW_ONLY
+      [[CDVWKInAppBrowser getInstance] loadAfterBeforeload:command];
+    #else
+      if(self.usewkwebview){
+          [[CDVWKInAppBrowser getInstance] loadAfterBeforeload:command];
+      }else{
+          [[CDVUIInAppBrowser getInstance] loadAfterBeforeload:command];
+      }
+    #endif
 }
 
 

--- a/src/ios/CDVUIInAppBrowser.h
+++ b/src/ios/CDVUIInAppBrowser.h
@@ -17,6 +17,8 @@
  under the License.
  */
 
+#if !WK_WEB_VIEW_ONLY
+
 #import <Cordova/CDVPlugin.h>
 #import <Cordova/CDVInvokedUrlCommand.h>
 #import <Cordova/CDVScreenOrientationDelegate.h>
@@ -89,3 +91,5 @@
 - (id)initWithUserAgent:(NSString*)userAgent prevUserAgent:(NSString*)prevUserAgent browserOptions: (CDVInAppBrowserOptions*) browserOptions;
 
 @end
+
+#endif

--- a/src/ios/CDVUIInAppBrowser.m
+++ b/src/ios/CDVUIInAppBrowser.m
@@ -17,6 +17,8 @@
  under the License.
  */
 
+#if !WK_WEB_VIEW_ONLY
+
 #import "CDVUIInAppBrowser.h"
 #import <Cordova/CDVPluginResult.h>
 #import <Cordova/CDVUserAgentUtil.h>
@@ -1126,4 +1128,4 @@ static CDVUIInAppBrowser* instance = nil;
 
 @end
 
-
+#endif

